libraries/ms-common/src/stm32f0xx/soft_timer.c

```diff
@@ -0,0 +1,95 @@
+#include "soft_timer.h"
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "status.h"
+#include "stm32f0xx_rcc.h"
+#include "stm32f0xx_tim.h"
+
+typedef struct timer {
```

Comment:
Just a general comment when designing data structures - be aware of [struct padding](http://www.catb.org/esr/structure-packing/).